### PR TITLE
Save/restore background

### PR DESCRIPTION
**Pull request by edpop**
User did not have to choose background again or retype custom

### CODE REVIEW

File: modules/renderer/background.js [hunk 1]
```diff
@@ -187,7 +188,7 @@ export function Background(context) {
         }
 
         var q = stringQs(location.hash.substring(1)),
-            chosen = q.background || q.layer,
+            chosen = q.background || q.layer || context.storage('background'),
```
Comment (bhousel) [inline modules/renderer/background.js]: The problem with putting it here is that now the user's preferred background might not be visible where they started editing the map.  This is especially a problem if a user is entering iD via a task manager or QA tool, and editing in a place that is not where they normally use iD.

Could you move the `context.storage` code down to the block where `best` gets set... Then in there set a variable called `recent` that contains this BackgroundSource only if it is valid in the current visible extent.

Then a few lines later do `background.baseLayerSource(findSource(chosen) || recent || best || ...etc)`

**Comment by edpop** [comment]
Ok, now there some much code.
On the other hand I can add parameter extent to "findSource" function if you like.

**Comment by edpop** [comment]
There is a wrong behaviour when we read a custom background from storage. We can't check it for extent. What should to do in this case?

File: modules/renderer/background.js [hunk 2]
```diff
@@ -202,13 +204,18 @@ export function Background(context) {
         backgroundSources.unshift(BackgroundSource.None());
 
         if (!chosen && extent) {
-            best = _.find(this.sources(extent), function(s) { return s.best(); });
+            var recent = findSource(context.storage('background'));
```
Comment (edpop) [inline modules/renderer/background.js]: context.storage('background') can be /^custom:/

**Comment by bhousel** [comment]
Nice work!  this will close #1756..
I think it's ok to restore the user's last used imagery, as long as it is valid where they are viewing the map.  (You're correct in your comment above that we can't do this validity check on a custom imagery source.)

**Pull request closed without merging**